geode-core/src/distributedTest/java/org/apache/geode/internal/cache/DeltaForceSizingFlagDUnitTest.java

```diff
@@ -0,0 +1,995 @@
+    });
+  }
+
+  private void assertValueType(VM vm, final Object key, final ValueType expectedType) {
+    vm.invoke("assertValueType", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
+      Object value = region.getValueInVM(key);
+      switch (expectedType) {
+        case RAW_VALUE:
+          assertTrue("Value was " + value + " type " + value.getClass(),
+              !(value instanceof CachedDeserializable));
+          break;
+        case CD_SERIALIZED:
+          assertTrue("Value was " + value + " type " + value.getClass(),
+              value instanceof CachedDeserializable);
+          assertTrue("Value not serialized",
+              ((CachedDeserializable) value).getValue() instanceof byte[]);
+          break;
+        case CD_DESERIALIZED:
+          assertTrue("Value was " + value + " type " + value.getClass(),
+              value instanceof CachedDeserializable);
+          assertTrue("Value was serialized",
+              !(((CachedDeserializable) value).getValue() instanceof byte[]));
+          break;
+        case EVICTED:
+          assertEquals(null, value);
+          break;
+      }
+    });
+  }
+
+  private static File[] getMyDiskDirs() {
+    long random = new Random().nextLong();
+    File file = new File(Long.toString(random));
+    file.mkdirs();
+    return new File[] {file};
+  }
+
+  private void createPR(MemberVM memberVM, final boolean enableLRU) {
+    memberVM.invoke("Create partitioned region", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      AttributesFactory<Integer, TestDelta> attr = new AttributesFactory<Integer, TestDelta>();
+      attr.setDiskSynchronous(true);
+      PartitionAttributesFactory<Integer, TestDelta> paf =
+          new PartitionAttributesFactory<Integer, TestDelta>();
+      paf.setRedundantCopies(1);
+      if (enableLRU) {
+        paf.setLocalMaxMemory(1); // memlru limit is 1 megabyte
+        attr.setEvictionAttributes(EvictionAttributes
+            .createLRUMemoryAttributes(new TestObjectSizer(), EvictionAction.OVERFLOW_TO_DISK));
+        attr.setDiskDirs(getMyDiskDirs());
+      }
+      PartitionAttributes<Integer, TestDelta> prAttr = paf.create();
+      attr.setPartitionAttributes(prAttr);
+      attr.setDataPolicy(DataPolicy.PARTITION);
+      attr.setSubscriptionAttributes(new SubscriptionAttributes(InterestPolicy.ALL));
+      Region<Integer, TestDelta> region = cache.createRegion("region", attr.create());
+    });
+
+  }
+
+  private void createRRHeapLRU(MemberVM memberVM) {
+    memberVM.invoke("Create rr-heap-lru", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      ResourceManager manager = cache.getResourceManager();
+      manager.setCriticalHeapPercentage(95);
+      manager.setEvictionHeapPercentage(90);
+      AttributesFactory<Integer, TestDelta> attr = new AttributesFactory<Integer, TestDelta>();
```

Comment:
`AttributesFactory` is deprecated and should not be used. `Cache.createRegionFactory()` should be used instead.